### PR TITLE
Link avatar login

### PR DESCRIPTION
**Pull request by guenk**
adding preview to avatar

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,5 +18,8 @@
   "devDependencies": {
     "concurrently": "^7.0.0",
     "husky": "^7.0.0"
+  },
+  "dependencies": {
+    "axios": "^1.2.2"
```
Comment (Sax593) [inline package.json]: Should not be here

File: frontend/src/pages/Home/Home.jsx [hunk 2]
```diff
@@ -17,6 +17,24 @@ export default function Home() {
       <section className="titlehome">
```
Comment (Sax593) [inline frontend/src/pages/Home/Home.jsx]: Already present on the dev

File: frontend/src/pages/Home/Style.scss [hunk 3]
```diff
@@ -43,7 +43,7 @@
```
Comment (Sax593) [inline frontend/src/pages/Home/Style.scss]: Already present on the dev

File: frontend/src/components/Myspace/Myspace.jsx [hunk 4]
```diff
@@ -17,6 +17,7 @@ export default function Myhome() {
       <Link id="home" className="menu-item" to="/">
```
Comment (Sax593) [inline frontend/src/components/Myspace/Myspace.jsx]: This component has rename and merge in dev. Please remove this!

File: frontend/src/components/Myspace/Style.scss [hunk 5]
```diff
@@ -108,3 +108,59 @@
   padding-top: 2.7em;
```
Comment (Sax593) [inline frontend/src/components/Myspace/Style.scss]: This component has rename and merge in dev. Please remove this!

File: frontend/src/App.jsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ function App() {
     <div className="App">
       <Routes>
         <Route path="/" element={<Home />} />
-        <Route path="/register" element={<SubscriptionForm />} />
+        <Route path="/sub" element={<SubscriptionForm />} />
```
Comment (Sax593) [inline frontend/src/App.jsx]: Please do not rename the route, /register is more meaningful than /sub

File: backend/package.json [hunk 7]
```diff
@@ -14,6 +14,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "axios": "^1.2.2",
```
Comment (Sax593) [inline backend/package.json]: Should not be here

**Review by Sax593 — CHANGES_REQUESTED** [review]

File: package.json [hunk 8]
```diff
@@ -20,6 +20,8 @@
     "husky": "^7.0.0"
   },
   "dependencies": {
-    "axios": "^1.2.2"
+    "argon2": "^0.30.3",
+    "axios": "^1.2.2",
+    "mock-aws-s3": "^4.0.2"
```
Comment (Sax593) [inline package.json:25]: Axios is a frontend dependencies, argon2 is backend dependencies.

File: backend/package.json [hunk 9]
```diff
@@ -14,7 +14,9 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "argon2": "^0.30.3",
```
Comment (Sax593) [inline backend/package.json]: Don't use, ca you delet this?

**Review by Sax593 — CHANGES_REQUESTED** [review]

**Review by Sax593 — APPROVED** [review]

**Review by Sax593 — DISMISSED** [review]

File: backend/src/models/UsersManager.js [hunk 10]
```diff
@@ -1,36 +1,44 @@
 const AbstractManager = require("./AbstractManager");
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
```
Comment (LoicBrassart) [inline backend/src/models/UsersManager.js]: -> `.env`

File: backend/src/models/UsersManager.js [hunk 11]
```diff
@@ -1,36 +1,44 @@
...
+const myPlaintextPassword = "s0//P4$$w0rD";
+const someOtherPlaintextPassword = "not_bacon";
```
Comment (LoicBrassart) [inline backend/src/models/UsersManager.js]: These variables are useless, aren't they?

Comment (LoicBrassart) [reply backend/src/models/UsersManager.js]: More, they are the example variables from the bcrypt package and are not used in this file

File: backend/src/models/UsersManager.js [hunk 12]
```diff
@@ -1,36 +1,44 @@
...
 
 class UsersManager extends AbstractManager {
   constructor() {
     super({ table: "users" });
   }
 
-  insert(users) {
+  async insert(users) {
+    const hashedPassword = await bcrypt.hash(users.password, 10);
```
Comment (LoicBrassart) [inline backend/src/models/UsersManager.js]: No hardcoded values

File: package.json [hunk 13]
```diff
@@ -18,5 +18,10 @@
   "devDependencies": {
     "concurrently": "^7.0.0",
     "husky": "^7.0.0"
+  },
+  "dependencies": {
+    "argon2": "^0.30.3",
+    "axios": "^1.2.2",
+    "mock-aws-s3": "^4.0.2"
```
Comment (LoicBrassart) [inline package.json:25]: And you shouldn't be using Amazon S3

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Nope, there are a lot of problems in this PR, how did you manage to commit it ? :raised_eyebrow:

**Review by Sax593 — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
sne

**Pull request merged**